core/src/main/java/org/apache/accumulo/core/iterators/WrappingIterator.java

```diff
@@ -58,21 +58,21 @@ protected void setSource(SortedKeyValueIterator<Key,Value> source) {
   public Key getTopKey() {
     if (seenSeek == false)
       throw new IllegalStateException("never been seeked");
-    return getSource().getTopKey();
+    return source.getTopKey();
```

Comment:
No, LightWrappingIterator is a user-level iterator, so it supports a no-argument constructor and non-final source to be compatible with the initialization via init method. It basically just removes some check, doesn't call getSource(), and exposes the source field as protected instead of private.